certbot-apache/certbot_apache/configurator.py

```diff
@@ -1377,7 +1377,20 @@ def enhance(self, domain, enhancement, options=None):
             raise errors.PluginError(
                 "Unsupported enhancement: {0}".format(enhancement))
         try:
-            func(self.choose_vhost(domain), options)
+            # Temp value tells choose_vhost to not to create a new SSL vhost if
+            # one is not found.
+            vhost = self.choose_vhost(domain, temp=True)
+            if not vhost.ssl:
+                msg_tmpl = ("Certbot was not able to find SSL VirtualHost for a "
+                       "domain {0} for enabling enhancement \"{1}\". The requested "
+                       "enhancement was not configured.")
+                if options:
+                    msg = msg_tmpl.format(domain, options)
+                else:
+                    msg = msg_tmpl.format(domain, enhancement)
+                logger.warning(msg)
```

Comment:
Why did you choose to just log a message here rather than raising an exception?

Reply:
There could be multiple domains to get enhanced, printing a clear error message tells the user that one of them has to be get fixed, but we'll be able to handle the rest of them.